### PR TITLE
Rockstar Games Files Added

### PR DESCRIPTION
**Pull request by bobstersok**
This is for the rockstar games website (https://www.rockstargames.com) and for subdomains like storerockstargames.com

### CODE REVIEW

File: websites/R/Rockstar Games/presence.ts [hunk 1]
```diff
@@ -0,0 +1,114 @@
+const presence = new Presence({
+    clientId: "765234467849240657"
+  }),
+  browsingTimestamp = Math.floor(Date.now() / 1000);
+  //strings = presence.getStrings({
+    //play: "presence.playback.playing",
+   // pause: "presence.playback.paused"
+    //You can use this to get translated strings in their browser language
+  //});
```
Comment (EncryptedDev) [inline websites/R/Rockstar Games/presence.ts:9]:
```suggestion
```

File: websites/R/Rockstar Games/presence.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+presence.on("UpdateData", async () => {
+  const presenceData: PresenceData = {
+    largeImageKey: "rockstargameshome",
+    startTimestamp: browsingTimestamp
+  },
+
+     {pathname} = document.location,
+     {hostname} = document.location;
```
Comment (EncryptedDev) [inline websites/R/Rockstar Games/presence.ts:18]:
```suggestion
  const presenceData: PresenceData = {
    largeImageKey: "rockstargameshome",
    startTimestamp: browsingTimestamp
  },
     {pathname, hostname} = document.location;
```

File: websites/R/Rockstar Games/presence.ts [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+  
+    if (hostname === "rockstargames.com" || hostname === "www.rockstargames.com") {
+      presenceData.largeImageKey = "rockstargameshome";
```
Comment (EncryptedDev) [inline websites/R/Rockstar Games/presence.ts:21]: There's no need in this if you have defined it on line 13 already
```suggestion
```

File: websites/R/Rockstar Games/presence.ts [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+      } else if (pathname === "/videos") 
+        presenceData.details = "Browsing Videos";
+       else if (pathname.startsWith("/videos/")) {
+        presenceData.details = "Browsing Videos";
+        presenceData.state = document.querySelector(
+          "head > title"
+        ).textContent;
+      } else if (pathname === "/downloads") 
+        presenceData.details = "Browsing Downloads";
+
```
Comment (EncryptedDev) [inline websites/R/Rockstar Games/presence.ts:53]: Remove empty line

File: websites/R/Rockstar Games/presence.ts [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+        else if (pathname.startsWith("/en/")) {
+        presenceData.details = "Browsing Rockstar Store";
+        presenceData.state = document.querySelector(
+          "head > title"
+        ).textContent;
+      } 
+    }
+    
+
+  if (presenceData.details === null) {
```
Comment (EncryptedDev) [inline websites/R/Rockstar Games/presence.ts:107]:
```suggestion
  if (!presenceData.details) {
```

File: websites/R/Rockstar Games/presence.ts [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+        presenceData.details = "Browsing Store Homepage";
+        else if (pathname.startsWith("/en/")) {
+        presenceData.details = "Browsing Rockstar Store";
+        presenceData.state = document.querySelector(
+          "head > title"
+        ).textContent;
+      } 
+    }
+    
+
```
Comment (EncryptedDev) [inline websites/R/Rockstar Games/presence.ts:106]:
```suggestion
```

**Review by EncryptedDev — CHANGES_REQUESTED** [review]

**Review by Bas950 — CHANGES_REQUESTED** [review]
upload proof of it working

**Comment by bobsterjsdev** [comment]
image
image
image

**Comment by bobsterjsdev** [comment]
This shows the presence of two different sites and shows when viewing a certain page

File: websites/R/Rockstar Games/dist/metadata.json [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+      "www.rockstargames.com",
+      "support.rockstargames.com",
+      "store.rockstargames.com",
+      "socialclub.rockstargames.com",
+      "signin.rockstargames.com"
+    ],
+    "description": {
+      "en": "Rockstar Games is a game development studio"
+    }
+  }
```
Comment (Slowlife01) [inline websites/R/Rockstar Games/dist/metadata.json:23]: `tags` is required

File: websites/R/Rockstar Games/presence.ts [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+      else if (pathname.startsWith("/en/")) {
+      presenceData.details = "Browsing Rockstar Store";
+      presenceData.state = document.querySelector(
+        "head > title"
+      ).textContent;
+    } 
+  }
+  
+if (!presenceData.details) {
+
```
Comment (EncryptedDev) [inline websites/R/Rockstar Games/presence.ts:98]: Empty lines
```suggestion
```

File: websites/R/Rockstar Games/presence.ts [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+    } 
+  }
+  
+if (!presenceData.details) {
+  
+  presence.setTrayTitle();
+  presence.setActivity(); 
+} else 
+  presence.setActivity(presenceData);
+
```
Comment (EncryptedDev) [inline websites/R/Rockstar Games/presence.ts:103]: Empty lines
```suggestion
```

**Review by EncryptedDev — CHANGES_REQUESTED** [review]
Minor refactoring issues, but otherwise seems fine for me

File: websites/R/Rockstar Games/presence.ts (hunk 9 above)
Comment (bobsterjsdev) [reply websites/R/Rockstar Games/presence.ts:103]: Ok, ill sort these issues out now

File: websites/R/Rockstar Games/dist/metadata.json [hunk 10]
```diff
@@ -0,0 +1,24 @@
+{
+    "$schema": "https://schemas.premid.app/metadata/1.3",
+    "service": "Rockstar Games",
+    "author": {
+      "name": "BobsterJS 💗",
+      "id": "359755774873960450"
+    },
+    "category": "games",
+    "logo": "https://i.imgur.com/7jcdG6X.png",
+    "thumbnail": "https://i.imgur.com/areSsgy.png",
```
Comment (Slowlife01) [inline websites/R/Rockstar Games/dist/metadata.json:10]: Better logo? And use the site screenshot instead

**Comment by bobsterjsdev** [comment]
I tried with a different logo and changed the thumbnail

File: websites/R/Rockstar Games/presence.ts [hunk 11]
```diff
@@ -0,0 +1,102 @@
...
+        "head > title"
+      ).textContent;
+    } 
+  }
+  
+  if (!presenceData.details) {
+    presence.setTrayTitle();
+    presence.setActivity(); 
+  } else 
+    presence.setActivity(presenceData);
```
Comment (EncryptedDev) [inline websites/R/Rockstar Games/presence.ts:101]:
```suggestion
  } else presence.setActivity(presenceData);
```

**Review by Slowlife01 — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**